CRM/Core/BAO/MessageTemplate.php

```diff
@@ -379,12 +381,16 @@ public static function sendTemplate($params) {
       }
     }
 
-    $subject = $dao->subject;
-    $text = $dao->text;
-    $html = $dao->html;
-    $format = $dao->format;
+    $mailContent = array(
+      'subject' => $dao->subject,
+      'text' => $dao->text,
+      'html' => $dao->html,
+      'format' => $dao->format,
+    );
     $dao->free();
 
+    CRM_Utils_Hook::alterMailContent($mailContent);
+
```

Comment:
I think all of the below variable assignment can be avoided if `$mailContent` is extracted with the `extract` keyword. Something like -

    extract($mailContent);

Can you please try this update and check if it works fine ?